### PR TITLE
Ide settings.md

### PR DESCRIPTION
**Pull request by hnwyllmm**
### Task Description

OceanBase is a huge C++ project and it's difficult to read the source code by using normal IDE. So, we need a recommend IDE and IDE settings document.

### Other Information
[preview](https://github.com/hnwyllmm/oceanbase/blob/ide_settings.md/docs/ide-settings.md)

### CODE REVIEW

**Comment by caifeizhi** [comment]
hi, @tisonkun can you review this PR?

File: docs/ide-settings.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+![ssh port](images/ide-settings-use-different-ssh-port.png)
+
+![ssh config file](images/ide-settings-choose-ssh-config.png)
+After that, the configured machines can be found in the `Connect to Host`. 
+
+Password need to be input everytime. If you want to skip this action, please configure SSH security login with credential. 
+
+## C/C++ Plugin
+
+C/C++ plugin can be download and installed in VSCode extension store in the case of simple scenarios:
```
Comment (caifeizhi) [inline docs/ide-settings.md:104]: it is an optional ? as C/C++ plugin is not compatible with ccls plugin

Comment (hnwyllmm) [reply docs/ide-settings.md:106]: When we write code with vscode, we can use c/c++ pulgin or ccls plugin, so it's an option.

I added our advice above this information.

File: docs/ide-settings.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# Abstract
+
+In order to easily read the code of OceanBase, we suggest using one IDE which is easily index the symbols of OceanBase. In Windows, we recommend `Souce Insight` can be used, and in Mac or Linux, we recommend that `VSCode + ccls` can be used to read the oceanbase code. Due to it is very easy to use `Source Ingisht`, so this document skip introduction how to use `Souce Insight`.
+
+This document introduce how to setup `VSCode + ccls`, which is very convenient to read the code of OceanBase. [ccls](https://github.com/MaskRay/ccls) is based on [cquery](https://github.com/jacobdufault/cquery), which is one of C/C++/Objective-C  [LSP](https://en.wikipedia.org/wiki/Language_Server_Protocol)s (In one word, LSP is used to provide programming language-specific features like code completion, syntax highlighting and marking of warnings and errors, as well as refactoring routines. ).
```
Comment (tisonkun) [inline docs/ide-settings.md:5]:
```suggestion
This document introduce how to setup `VSCode + ccls`, which is very convenient to read the code of OceanBase. [ccls](https://github.com/MaskRay/ccls) is based on [cquery](https://github.com/jacobdufault/cquery), which is one of C/C++/Objective-C [LSP](https://en.wikipedia.org/wiki/Language_Server_Protocol)s (In one word, LSP is used to provide programming language-specific features like code completion, syntax highlighting and marking of warnings and errors, as well as refactoring routines. ).
```

**Pull request merged**